File: app/src/main/java/com/github/jiahuanyu/example/servicedialog/MyService.java
package com.github.jiahuanyu.example.servicedialog;

import android.app.Service;
import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.content.IntentFilter;
import android.os.IBinder;
import android.support.annotation.Nullable;
import android.view.WindowManager;

import com.github.jiahuanyu.example.R;

/**
 * Created by doom on 16/3/30.
 */
public class MyService extends Service
{
    private android.app.AlertDialog mCommonAlertDialog;
    private android.support.v7.app.AlertDialog mV7Dialog;
    private BroadcastReceiver mReceiver = new BroadcastReceiver()
    {
        @Override
        public void onReceive(Context context, Intent intent)
        {
            String action = intent.getAction();
            if (ServiceDialogActivity.SHOW_COMMON_DIALOG_ACTION.equals(action))
            {
                showCommonDialog();
            }
            else if (ServiceDialogActivity.SHOW_V7_DIALOG_ACTION.equals(action))
            {
                showV7Dialog();
            }
        }
    };


    private void showCommonDialog()
    {
        if (mCommonAlertDialog == null)
        {
            mCommonAlertDialog = new android.app.AlertDialog.Builder(getApplicationContext())
                    .setPositiveButton("确定", null)
                    .setTitle("普通AlertDialog_Title")
                    .setMessage("普通AlertDialog_Message")
                    .create();
            mCommonAlertDialog.getWindow().setType((WindowManager.LayoutParams.TYPE_SYSTEM_ALERT));
        }
        if (!mCommonAlertDialog.isShowing())
        {
            mCommonAlertDialog.show();
        }
    }

    private void showV7Dialog()
    {
        if (mV7Dialog == null)
        {
            mV7Dialog = new android.support.v7.app.AlertDialog.Builder(getApplicationContext(), R.style.AppTheme_AlertDialog)
                    .setPositiveButton("确定", null)
                    .setTitle("V7包AlertDialog_Title")
                    .setMessage("V7包AlertDialog_Message")
                    .create();
            mV7Dialog.getWindow().setType((WindowManager.LayoutParams.TYPE_SYSTEM_ALERT));
        }
        if (!mV7Dialog.isShowing())
        {
            mV7Dialog.show();
        }
    }

    @Override
    public void onCreate()
    {
        super.onCreate();
        IntentFilter filter = new IntentFilter();
        filter.addAction(ServiceDialogActivity.SHOW_COMMON_DIALOG_ACTION);
        filter.addAction(ServiceDialogActivity.SHOW_V7_DIALOG_ACTION);
        registerReceiver(mReceiver, filter);
    }

    @Nullable
    @Override
    public IBinder onBind(Intent intent)
    {
        return null;
    }

    @Override
    public void onDestroy()
    {
        super.onDestroy();
        unregisterReceiver(mReceiver);
    }
}

File: app/src/main/java/me/jiahuan/androidlearn/service/MyService.kt
package me.jiahuan.androidlearn.service


import android.app.Service
import android.content.Intent
import android.os.Binder
import android.os.IBinder
import android.util.Log

class MyService : Service() {

    private val mBinder = MyBinder()

    override fun onCreate() {
        Log.d(TAG, "onCreate")
        super.onCreate()
    }

    override fun onStartCommand(intent: Intent, flags: Int, startId: Int): Int {
        Log.d(TAG, "onStartCommand")
        return super.onStartCommand(intent, flags, startId)
    }


    override fun onDestroy() {
        Log.d(TAG, "onDestroy")
        super.onDestroy()
    }

    override fun onRebind(intent: Intent) {
        Log.d(TAG, "onRebind")
        super.onRebind(intent)
    }

    override fun onUnbind(intent: Intent): Boolean {
        Log.d(TAG, "onUnbind")
        return super.onUnbind(intent)
    }

    override fun onBind(intent: Intent): IBinder? {
        Log.d(TAG, "onBind")
        return mBinder
    }


    internal inner class MyBinder : Binder() {

        val service: MyService
            get() = this@MyService

    }


    fun log() {
        Log.d(TAG, "log in service")
    }

    companion object {

        private val TAG = "MyService"
    }
}
